scalafix-testkit/src/main/scala/scalafix/testkit/CommentAssertion.scala

```diff
@@ -5,90 +5,79 @@ import scala.meta._
 import scala.util.matching.Regex
 import org.langmeta.internal.ScalafixLangmetaHacks
 
-sealed trait CommentAssertion {
-  def position: Position
-  def key: String
-  def formattedMessage: String
-}
-object CommentAssertion {
-  def extract(tokens: Tokens): List[CommentAssertion] = {
-    tokens.collect {
-      case EndOfLineAssertExtractor(singleline) =>
-        singleline
-
-      case MultiLineAssertExtractor(multiline) =>
-        multiline
-    }.toList
-  }
-}
-
-// EndOfLineAssert are the bread and butter of testkit they
+// CommentAssertion are the bread and butter of testkit they
 // assert the line position and the category id of the lint message.
 //
 // For example:
 //
 // ```scala
 // Option(1).get // assert: Disable.get
 // ```
-case class EndOfLineAssert(anchorPosition: Position, key: String)
-    extends CommentAssertion {
-  def position: Position = anchorPosition
-
-  def formattedMessage: String =
-    ScalafixLangmetaHacks.formatMessage(
-      anchorPosition,
-      "error",
-      ""
-    )
-}
-object EndOfLineAssertExtractor {
-  val AssertRegex: Regex = " assert: (.*)".r
-  def unapply(token: Token): Option[EndOfLineAssert] = {
-    token match {
-      case Token.Comment(AssertRegex(key)) =>
-        Some(EndOfLineAssert(token.pos, key))
-      case _ =>
-        None
-    }
-  }
-}
-
-// MultiLineAssert are useful two visually show the lint message
-// it adds an assertion on the message body and the carret position
-// of the lint message.
+//
+// You can also use the multiline variant. This isuseful two visually
+// show the lint message it adds an assertion on the message body and
+// the carret position of the lint message.
 //
 // For example:
 //
 // ```scala
 // Option(1).get /* assert: Disable.get
 //           ^
-//
 // Option.get is the root of all evilz
 //
 // If you must Option.get, wrap the code block with
 // // scalafix:off Option.get
 // ...
 // // scalafix:on Option.get
 // */
-case class MultiLineAssert(
+case class CommentAssertion(
     anchorPosition: Position,
-    carretPosition: Position,
     key: String,
-    expectedMessage: String)
-    extends CommentAssertion {
-  def position: Position = anchorPosition
+    carretPosition: Option[Position],
+    expectedMessage: Option[String]) {
 
   def formattedMessage: String =
     ScalafixLangmetaHacks.formatMessage(
-      carretPosition,
+      carretPosition.getOrElse(anchorPosition),
       "error",
-      "\n" + expectedMessage
+      expectedMessage.map("\n" + _).getOrElse("")
     )
 }
+
+object CommentAssertion {
+  def extract(tokens: Tokens): List[CommentAssertion] = {
+    tokens.collect {
+      case EndOfLineAssertExtractor(singleline) =>
```

Comment:
Nitpick, I try to make my extractors as specific as possible and compose them so that we could something like this
```
case Token.Comment(CommentAssert(assert)) => assert

object CommentAssert {
  def unapply(arg: String): Option[CommentAssert] = MultilineAssert.unapply(arg).orElse(EndOfLineAsssert.unapply(arg))
}
object MultilineAssert {
  def unapply(arg: String) = arg match { case AssertRegex(key) => ... }
}
...
```